cfpb/consumerfinance.gov
# Add separate field for custom navigation label

Right now, the labels for items within secondary navigation (i.e. Browse page navigation) are determined solely by the Wagtail title field. This is undesirable for a few reasons:

- The current functionality is not explicit or described in Wagtail.
- The use of the Wagtail title field and the use of the secondary navigation are not identical. Secondary navigation items often use shortened versions of names for better readability and usability of the navigation field, but these shortened names can cause problems with searchability within the Wagtail admin panel.

This PR creates a new field, `navigation_label`, which can be used optionally as a custom shortened title for navigation items. Logic is included so that secondary navigation _defaults_ to using the Wagtail title if this field is blank.

---




## Additions

A new field, `navigation_label`, editable within any page's "Sidebar" tab:

![Screen Shot 2023-02-22 at 10 31 59 AM](https://user-images.githubusercontent.com/29648039/220671599-c53e64ed-893f-404c-8b40-d2117e7dc387.png)


## Changes

- Logic has been added to `v1/util/util.py` to populate the field with the Wagtail title field if this new field is blank.


## How to test this PR

1. localhost
2. Navigate to a Browse page that appears within secondary navigation (e.g. `localhost/enforcement/enforcement-by-the-numbers/`
3. In the page editor, select the Sidebar tab and fill in a new title under "Navigation Label"
4. Publish the page and confirm that the new label now appears within secondary navigation for all pages in the section


## Screenshots
![Screen Shot 2023-02-22 at 10 31 59 AM](https://user-images.githubusercontent.com/29648039/220671599-c53e64ed-893f-404c-8b40-d2117e7dc387.png)
![Screen Shot 2023-02-22 at 10 34 19 AM](https://user-images.githubusercontent.com/29648039/220672488-df29997d-f823-456e-a7ff-e7ffb679c1a4.png)


## Notes and todos

- I decided to have this on *all* page types rather than just Browse pages because a) it removes the need for duplication on both Browse and Browse filterable list page types, and b) I'm not 100% certain we don't have a case where a non-Browse page would still need to appear within secondary navigation.


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
